### PR TITLE
Allows region/service to be specified explicitly, skips magic

### PR DESCRIPTION
**Pull request by crunchytom**
Jira: CXCMS-1170
Reviewer: epurkhiser

### CODE REVIEW

**Comment by crunchytom** [comment]
this is back-compatible (and, you were right, only took a few minutes)

File: sign4.go [hunk 1]
```diff
@@ -57,6 +57,13 @@ func stringToSignV4(request *http.Request, hashedCanonReq string, meta *metadata
...
 
 	meta.algorithm = "AWS4-HMAC-SHA256"
+	service, region := serviceAndRegion(request.Host)
+	if meta.service == "" {
+		meta.service = service
+	}
+	if meta.region == "" {
+		meta.region = region
+	}
 	meta.service, meta.region = serviceAndRegion(request.Host)
```
Comment (pdunna2) [inline sign4.go]: line 67 redoing  previous assignments inside if conditions, is it intentional?

Comment (crunchytom) [inline sign4.go]: that's an oops, good catch.

**Comment by pdunna2** [comment]
👍

**Pull request merged**